.infrastructure/helm-chart/todoapp/templates/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secret
+  namespace: {{ .Values.todoapp.namespace }}
+type: Opaque
+data:
+  {{- range $k, $v := .Values.todoapp.secrets}}
+  {{ $k }}: {{ $v }}
+  {{- end }}
```

Comment:
Ensure that the `todoapp.secrets` values are defined in the `values.yaml` file. This is necessary for populating the Secret data correctly.